### ISSUE TITLE
[BUG]: Runs recorded in MLflow nests all recursively when [full] installed

### ISSUE THREAD

**Issue by nagamatz** [pycaret/pycaret#3059]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [ ] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

When pycaret is installed with [full], all runs executed in one script are shown nested recursively in MLflow dashboard.
This happens only with [full] installation.

### Reproducible Example

```python
%pip install -U pip wheel
%pip install --pre pycaret[full]

import mlflow
from pycaret.classification import *
from pycaret.datasets import get_data

mlflow.set_tracking_uri("http://localhost:5000")

data = get_data("diabetes")
setup(data, target="Class variable", log_experiment=True)
compare_models(include=["lr", "svm", "rf"])
setup(data, target="Class variable", log_experiment=True)
compare_models(include=["lr", "svm", "rf"])
```

### Expected Behavior

Expected display: (when installed without [full])
![OK](https://user-images.githubusercontent.com/1991802/198862894-7a459755-5b94-4abc-a00b-be8d42e1f71c.png)

Actual display: (when installed with [full])
![NG](https://user-images.githubusercontent.com/1991802/198862906-a26034b1-e22b-4d36-a0e5-1f0c5ccdad8c.png)

### Actual Results

```python-traceback
Attached the figure also in 'Expected Behavior'.
```

### Installed Versions

<details>
System:
    python: 3.9.5 (default, Nov 23 2021, 15:27:38)  [GCC 9.3.0]
executable: /home/ak/sample/.venv/bin/python
   machine: Linux-5.10.102.1-microsoft-standard-WSL2-x86_64-with-glibc2.31

PyCaret required dependencies:
                 pip: 22.3
          setuptools: 44.0.0
             pycaret: 3.0.0rc4
             IPython: 8.5.0
          ipywidgets: 8.0.2
                tqdm: 4.64.1
               numpy: 1.22.4
              pandas: 1.4.4
              jinja2: 3.1.2
               scipy: 1.8.1
              joblib: 1.2.0
             sklearn: 1.1.3
                pyod: 1.0.6
            imblearn: 0.9.1
   category_encoders: 2.5.1.post0
            lightgbm: 3.3.3
               numba: 0.55.2
            requests: 2.28.1
          matplotlib: 3.5.3
          scikitplot: 0.3.7
         yellowbrick: 1.5
              plotly: 5.11.0
             kaleido: 0.2.1
         statsmodels: 0.13.2
              sktime: 0.13.4
               tbats: 1.1.1
            pmdarima: 1.8.5
              psutil: 5.9.3

PyCaret optional dependencies:
                shap: 0.41.0
           interpret: 0.2.7
                umap: 0.5.3
    pandas_profiling: 3.4.0
  explainerdashboard: 0.4.0
             autoviz: 0.1.58
           fairlearn: 0.8.0
             xgboost: 1.7.0rc1
            catboost: 1.1
              kmodes: 0.12.2
             mlxtend: 0.21.0
       statsforecast: 1.1.3
        tune_sklearn: 0.4.4
                 ray: 2.0.1
            hyperopt: 0.2.7
              optuna: 3.0.3
               skopt: 0.9.0
              mlflow: 1.30.0
              gradio: 3.8
             fastapi: 0.85.1
             uvicorn: 0.19.0
              m2cgen: 0.10.0
           evidently: 0.1.59.dev2
                nltk: 3.7
            pyLDAvis: Not installed
              gensim: Not installed
               spacy: Not installed
           wordcloud: 1.8.2.2
            textblob: 0.17.1
               fugue: 0.6.6
           streamlit: Not installed
             prophet: Not installed
</details>

**Comment by tdekelver-bd** [comment]
In addition, runs of `compare_models `and `create_model` get nested recursively as well in pycaret > 2.3.6.
See screenshot below where the red line shows the behaviour in pycaret==2.3.6 (which is the wanted and expected behaviour) and in orange the nested unwanted behaviour in pycaret 2.3.8 , 2.3.9 and 2.3.10
![image](https://user-images.githubusercontent.com/50994394/200543740-0883c8ba-9f4a-4d1f-8abc-560ae7dbd54e.png)

**Comment by moezali1** [comment]
@nagamatz @tdekelver-bd This issue was recently fixed on last rc release. Can you try installing pycaret with `pip install --pre pycaret` and let me know if you still face the issue.

**Comment by nagamatz** [comment]
It's not yet fixed with 3.0.0rc4. This is the results with the code in the first post. Now, mlflow is 2.0.1
![無題](https://user-images.githubusercontent.com/1991802/206426156-4b19a4cc-b865-4af3-8281-1b89fc099f28.png)
